src/lists.rs

```diff
@@ -41,11 +47,38 @@ pub struct ListFormatting<'a> {
     pub h_width: usize,
     // Available width if we layout vertically
     pub v_width: usize,
+    // Non-expressions, e.g. items, will have a new line at the end of the list.
+    // Important for comment styles.
+    pub is_expression: bool
```

Comment:
I don't want to worry too much about this - it should be a total edge case exactly because there is no way to format it nicely. Another alternative might be to change all post-comments to /* */. Or to only do that if there is a final comment? Any way we do this is kind of inconsistent. I guess I prefer a little inconsistency in the punctuation rather than in the comment style (and the code should be simpler), but I don't think it is too important.